torch/csrc/jit/script/compiler.cpp

```diff
@@ -2366,6 +2366,15 @@ struct to_ir {
     } else {
       auto inputs = getNamedValues(apply.inputs(), true);
       auto attributes = emitAttributes(apply.attributes());
+
+      auto simple_value = std::dynamic_pointer_cast<SimpleValue>(sv);
```

Comment:
I think this should go in the `call` implementation in SimpleValue. In general, we should consider dynamic_casting an anti-pattern in this part of the code.

Reply:
That doesn't work, because you have to add the classtype value as the `self` argument

Reply:
Not sure I understand—why can't you do that in `SimpleValue::call()`? You have access to `self` there.

Reply:
Ahh got it to work. I tried the way you suggested previously but I forgot to emplace the `self argument`.